core/src/main/java/io/crate/settings/CrateSetting.java

```diff
@@ -66,4 +66,12 @@ public String getKey() {
     public T getDefault() {
         return setting.getDefault(Settings.EMPTY);
     }
+
+    public boolean isGroupSetting() {
+        /**
+         * there is already a method called isGroupSetting in ES Setting class
+         * however it is package private.
+         */
+        return setting.getKey().endsWith(".");
```

Comment:
pls do this while constructing the instance and use a boolean which will be return here, so it won't be execute every time one calls isGroupSetting()